src/rulesets/LinearAlgebra/matfun.jl

```diff
@@ -129,7 +129,10 @@ function rrule(::typeof(exp), A0::StridedMatrix{<:BlasFloat})
         A = copy(A0)
         X, intermediates = _matfun!(exp, A)
         function exp_pullback(ΔX)
-            ∂A = _matfun_frechet_adjoint!(exp, ΔX, A, X, intermediates)
+            # Ensures ∂X is mutable. The outer `adjoint` is unwrapped without copy by
+            # the default _matfun_frechet_adjoint!
+            ∂X = ChainRulesCore.is_inplaceable_destination(ΔX) ? ΔX : convert(Matrix, ΔX')'
```

Comment:
Why not just:
```suggestion
            ∂X = ChainRulesCore.is_inplaceable_destination(ΔX) ? ΔX : convert(Matrix, ΔX)
```

Reply:
Because if `∂X'` is an `Adjoint`, then `_matfun_frechet_adjoint!` will `copy` it to make it non-`Adjoint`. This way, we do only one allocation instead of 2. An alternative is to bypass `_matfun_frechet_adjoint!` to call `_matfun_frechet!` directly, but to me this seems cleaner. What do you think?

Reply:
But if it is an `Adjoint` then isn't it going to be mutable?
Or not becuase it might be an `Adjoint{FillArray}` ?

Reply:
I guess this makes sense, leave a comment to that effect?

Reply:
Can you suggest a change that would make this comment clearer?
https://github.com/JuliaDiff/ChainRules.jl/blob/608f3af380fe8d7a20085b0d57f2d9ea2e8a2e00/src/rulesets/LinearAlgebra/matfun.jl#L132-L133